File: Projects/Project_03_1D_Burger/For_Instructors/julia/element_matrices.jl
#---------------------------------------------------------------------#
#This code computes the element mass and differentiation matrices
#Written by F.X. Giraldo on April 24, 2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function element_matrices(ψ,dψ,Np,Nq,ωq,DFloat)

    #Initialize Matrices
    M=zeros(DFloat,Np,Np)
    D=zeros(DFloat,Np,Np)
    Dw=zeros(DFloat,Np,Np)
    F=zeros(DFloat,Np,Np)

    #Construct Mass and Differentiation Matrices
    @inbounds for k=1:Nq, j=1:Np, i=1:Np
        M[i,j]+=ωq[k]*ψ[i,k]*ψ[j,k]
        D[i,j]+=ωq[k]*ψ[i,k]*dψ[j,k]
        Dw[i,j]+=ωq[k]*dψ[i,k]*ψ[j,k]
    end
    F[1,1]=-1; F[Np,Np]=+1

    return (M,D,Dw,F)
end

File: Projects/Project_04_1D_SWE/Old/For_Students/matlab/element_matrices.m
%---------------------------------------------------------------------%
%This function computes the LGL grid and elements.
%Written by F.X. Giraldo on 10/2003
%           Department of Applied Mathematics
%           Naval Postgraduate School
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function [Me, Dwe, Fe] = element_matrices(psi,dpsi,ngl,nq,wnq)

%Initialize
Me=zeros(ngl,ngl);
Dwe=zeros(ngl,ngl);
Fe=zeros(ngl,ngl);
 
%Do LGL Integration
for l=1:nq
  wq=wnq(l);
  for j=1:ngl
     for i=1:ngl
        Me(i,j)=Me(i,j) + wq*psi(i,l)*psi(j,l);
        Dwe(i,j)=Dwe(i,j) + wq*dpsi(i,l)*psi(j,l);
     end %j
  end %i
end %l
Fe(1,1)=-1;
Fe(ngl,ngl)=+1;



      
